### PR TITLE
Dockerfile

### LINKED ISSUE #23: Dockerfile
Im trying to dockerize this project but I'm a bit at loss at how to deal with registration file generation. Right now my best bet is to do it in the image and have the user mount a dir to access it. Feels wrong but generating it outside of the container means depending on node which kinda defeats the purpose..

You got any ideas?

### PR DESCRIPTION
**Pull request by joenas**
Fixes #23 

Not really much to add other than that I tried to make the README as explicit as possible. Maybe move it to something like `DOCKER.md` and link it from the readme?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,3 +69,63 @@ Invite the webhook bridge to your room (`@_webhook:t2bot.io`) and send the messa
...
+
+There are two gotchas when running with Docker.
+
+First of all the appservice needs to communicate with synapse and there are two ways. Either create a docker network and set the urls accordingly or use public urls (ie webhooks.example.com or example.com:9000). Using `localhost:9000` as the url for appservice won't work because synapse has it's "own" localhost.
+
+Secondly, you need to generate the registration file and let synapse (or whatever server you're using) have access to it. The approach here is to generate it with a temporary container and a [bind mount](https://docs.docker.com/engine/admin/volumes/bind-mounts/) so the file is accessible on the host and can be copied to the homeserver.
+
+Below are a few example of how do achieve this. There is also a `docker-compose.example.yaml`.
+
+**Note** I've had some trouble getting synapse to properly talk to the appservice on the docker network (the reverse works fine). If you experience something similar, expose `9000` on the host or reverse proxy it and use the public url instead of `http://matrix_webhooks_container:9000` in the examples.
```
Comment (turt2live) [inline README.md]: Port 9000 should absolutely never be exposed to the public internet. It's an unencrypted, unsecured, channel for injecting events to the application service.

Comment (joenas) [reply README.md]: That makes sense. Do you consider a reverse proxy ok or should I remove that part completely?

Still trying to figure out why it not reliably working with docker network both ways.

Comment (turt2live) [reply README.md]: It should be internal-only. Ideally localhost if possible, but otherwise a private network.

Comment (joenas) [reply README.md]: Alright I figured it out. The hostname of the appservice container in the docker network wasn't valid according to synapse (prolly makes sense)

```
synapse.appservice.api - 228 - WARNING - - push_bulk to http://matrix_webhooks_1:9000/transactions/1 threw exception invalid hostname: matrix_webhooks_1
```

Changing to a valid hostname/container name fixes it and there is no reason to expose `9000` outside the host or even the docker network, which is only accessible to containers connected to it!

**Comment by turt2live** [comment]
I'll have to take more in-depth look and test this a bit later, but overall it looks good.

**Pull request merged**